newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js

```diff
@@ -161,17 +164,28 @@ const AnimationPreview = ({
 
         currentFrameIndexRef.current = newFrameIndex;
         currentFrameElapsedTimeRef.current = newFrameElapsedTime;
-        // Ensure we trigger an update if the animation changes,
+        const newResourceName = resourceNames[currentFrameIndexRef.current];
+        // Ensure we trigger an update if the frame changes,
         // as the refs will not do it.
         if (currentFrameIndex !== newFrameIndex) {
-          imagesLoadedArray.current[currentFrameIndexRef.current] = false;
-          // When the array of loaders changes, wait a bit to display the loader to avoid flickering.
-          loaderTimeout.current = setTimeout(() => {
-            console.warn(
-              'The image took too long to load, displaying a loader.'
-            );
-            setIsStillLoadingResources(true);
-          }, 500);
+          if (newResourceName === currentResourceNameRef.current) {
```

Comment:
Shouldn't it be the path rather than the name of the file?

Reply:
Tried that, but:
- there is a one to one relation between name & path (and a function is called to get the path from the name) so the result should be the same
- but when a file is cached locally, it adds a `?cache={date}` in the url, so the urls actually differ as the animation continues, but the browser is smart enough to used the cached image and not trigger the onLoad.

tldr; I tried but it felt more complex to make it work

Reply:
Are you sure there is a one to one relation between names and paths for resources? Because today you can have multiple objects using the same resource.

Reply:
I think this is different here, this is in the scope of 1 object & 1 animation only.
The animation knows where to find the resource from the name of this resource.
It uses the function `resourcesLoader.getResourceFullUrl(project, name, {})` in the `DirectionTools`